File: SDKs/twilio.com/twilio_preview/1.36.0/go/pkg/models/operations/updateunderstandquery.go
package operations

import (
	"openapi/pkg/models/shared"
)

var UpdateUnderstandQueryServers = []string{
	"https://preview.twilio.com",
}

type UpdateUnderstandQueryPathParams struct {
	AssistantSid string `pathParam:"style=simple,explode=false,name=AssistantSid"`
	Sid          string `pathParam:"style=simple,explode=false,name=Sid"`
}

type UpdateUnderstandQueryRequestBodyUpdateUnderstandQueryRequest struct {
	SampleSid *string `form:"name=SampleSid"`
	Status    *string `form:"name=Status"`
}

type UpdateUnderstandQuerySecurity struct {
	AccountSidAuthToken shared.SchemeAccountSidAuthToken `security:"scheme,type=http,subtype=basic"`
}

type UpdateUnderstandQueryRequest struct {
	ServerURL  *string
	PathParams UpdateUnderstandQueryPathParams
	Request    *UpdateUnderstandQueryRequestBodyUpdateUnderstandQueryRequest `request:"mediaType=application/x-www-form-urlencoded"`
	Security   UpdateUnderstandQuerySecurity
}

type UpdateUnderstandQueryResponse struct {
	ContentType                     string
	StatusCode                      int64
	PreviewUnderstandAssistantQuery *shared.PreviewUnderstandAssistantQuery
}

File: SDKs/twilio.com/twilio_preview/1.37.3/python/src/sdk/models/operations/updateunderstandquery.py
from dataclasses import dataclass, field
from datetime import date, datetime
from marshmallow import fields
import dateutil.parser
from typing import Optional
from sdk.models import shared


UPDATE_UNDERSTAND_QUERY_SERVERS = [
	"https://preview.twilio.com",
]


@dataclass
class UpdateUnderstandQueryPathParams:
    assistant_sid: str = field(metadata={'path_param': { 'field_name': 'AssistantSid', 'style': 'simple', 'explode': False }})
    sid: str = field(metadata={'path_param': { 'field_name': 'Sid', 'style': 'simple', 'explode': False }})
    

@dataclass
class UpdateUnderstandQueryUpdateUnderstandQueryRequest:
    sample_sid: Optional[str] = field(default=None, metadata={'form': { 'field_name': 'SampleSid' }})
    status: Optional[str] = field(default=None, metadata={'form': { 'field_name': 'Status' }})
    

@dataclass
class UpdateUnderstandQuerySecurity:
    account_sid_auth_token: shared.SchemeAccountSidAuthToken = field(metadata={'security': { 'scheme': True, 'type': 'http', 'sub_type': 'basic' }})
    

@dataclass
class UpdateUnderstandQueryRequest:
    path_params: UpdateUnderstandQueryPathParams = field()
    security: UpdateUnderstandQuerySecurity = field()
    request: Optional[UpdateUnderstandQueryUpdateUnderstandQueryRequest] = field(default=None, metadata={'request': { 'media_type': 'application/x-www-form-urlencoded' }})
    server_url: Optional[str] = field(default=None)
    

@dataclass
class UpdateUnderstandQueryResponse:
    content_type: str = field()
    status_code: int = field()
    preview_understand_assistant_query: Optional[shared.PreviewUnderstandAssistantQuery] = field(default=None)
    
